gluonnlp/vocab.py

```diff
@@ -77,6 +81,8 @@ class Vocab(object):
         The embedding of the indexed tokens.
     idx_to_token : list of strs
         A list of indexed tokens where the list indices and the token indices are aligned.
+    idx_to_counts : numpy.ndarray
```

Comment:
Do we really want to couple vocab with counts? We already have Counter. Is vocab now also responsible for maintaining the counts for all tokens?

Reply:
We need the combination of index and counts. But you're right, it may be better to construct that on the fly by taking vocab and counter as an argument where it is needed.